niceman/resource/base.py

```diff
@@ -27,12 +28,18 @@
 from ..support.exceptions import MissingConfigError, MissingConfigFileError
 from ..ui import ui
 
-# Enumerate the defined resource types available.
-VALID_RESOURCE_TYPES = [
-    'docker-container',
-    'aws-ec2',
-    'shell'
-]
+def attrib(*args, **kwargs):
+    """
+    Extend the attrs decorator to include a doc metadata element.
+    """
+    doc = kwargs.get('doc')
+    metadata = kwargs.get('metadata', {})
+    if doc:
+        metadata['doc'] = doc
+        del kwargs['doc']
```

Comment:
Nit picking: You could just .pop it (instead of get) I believe giving it a default None this avoiding del